### DISCUSSION TITLE
Airflow multiple instances with common DB

### DISCUSSION THREAD

**Discussion by nitra-1** [apache/airflow#26154] [General] [answered]
Hi,
What will be the behaviour of the Airflow 2.0.2 if there are multiple instances each with LocalExecutor, having some common DAGs, and pointing to same metadata DB(PG).
Will the scheduler on each instance run the same DAGs? or any other impact ?

**Comment by potiuk** [comment, marked as answer]
If you use Postgres, it should work with no problems theorethicaly. but if you have some experiences with that you will be able to share, that would be fantastic. This is a multi-scheduler setting we anticipated it should work but never heard anyone using it

BTW. you should miggrate to latest released airflow. There were probably more than 1000 bugfixes since 2.0.2 so you might experience one of those 1000 bugs in 2.0.2 and you might hit one of those. By upgrading you save yourself (and those who you reach out to) time for diagnosing those and finding they were already fixed.

By not upgrading you are also making yourself vulnerable to a number of security issues fixed since. It's a very bad idea

It's pretty unwise to stay wth 2.0.2. Airflow uses SemVer as of Airflow 2 so upgrading to latest 2.* release is backwards compatible.

**Reply by nitra-1** [reply]
Hi, Thank you for the insight, will definitely update this thread with my experience.
Does upgrade mean re-installation of latest version or are there some upgrade scripts, would be helpful if you could point me to some document.

**Reply by potiuk** [reply]
Just look for "upgrading" in airflow docs. It has all the explanation.

**Reply by joshmarkovic** [reply]
@nitra-1 can you please update us on your setup? Having Multiple localexecutors is a very interesting setup!

**Comment by nitra-1** [comment]
I wanted a backup server in case something goes wrong with one server.
Both were pointing to same PG.
Scheduler was running on both servers.
Tried this setup for few days but did not go ahead with that as that
project was abandoned.
My observation was that request would execute on only one server.But could
not test the scenario when one airflow goes down.
